contrib/avro/src/python/pants/contrib/avro/tasks/avro_gen.py

```diff
@@ -0,0 +1,93 @@
+# coding=utf-8
+# Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+
+from pants.backend.jvm.targets.java_library import JavaLibrary
+from pants.backend.jvm.tasks.nailgun_task import NailgunTask
+from pants.base.exceptions import TaskError
+from pants.build_graph.address import Address
+from pants.build_graph.address_lookup_error import AddressLookupError
+from pants.java.jar.jar_dependency import JarDependency
+from pants.task.simple_codegen_task import SimpleCodegenTask
+from twitter.common.collections import OrderedSet
+
+from pants.contrib.avro.targets.java_avro_library import JavaAvroLibrary
+
+
+AVRO_REV = '1.8.2'
+
+
+class AvroJavaGenTask(SimpleCodegenTask, NailgunTask):
+  """Compile Avro schema, protocol, and IDL files to Java code."""
+
+  @classmethod
+  def register_options(cls, register):
+    super(AvroJavaGenTask, cls).register_options(register)
+    # pylint: disable=no-member
+    cls.register_jvm_tool(register,
+                          'avro-tools',
+                          classpath=[JarDependency(org='org.apache.avro', name='avro-tools', rev=AVRO_REV)],)
+    register('--runtime-deps', advanced=True, type=list, fingerprint=True,
+             default=['//:avro-java-runtime'],
+             help='A list of specs pointing to dependencies of Avro generated code.')
+
+  def __init__(self, *args, **kwargs):
+    super(AvroJavaGenTask, self).__init__(*args, **kwargs)
+
+  def synthetic_target_type(self, target):
+    return JavaLibrary
+
+  def synthetic_target_extra_dependencies(self, target, target_workdir):
+    dependencies = OrderedSet()
+    for dep_spec in self.get_options().runtime_deps:
+      dep_address = Address.parse(dep_spec)
+      try:
+        self.context.build_graph.maybe_inject_address_closure(dep_address)
+        dependencies.add(self.context.build_graph.get_target(dep_address))
+      except AddressLookupError as e:
+        raise AddressLookupError('{}\n  referenced from {} scope'.format(e, self.options_scope))
+
+    dependencies.update(target.dependencies)
+    return dependencies
+
+  def synthetic_target_extra_exports(self, target, target_workdir):
```

Comment:
Exporting all of your dependencies is not a good default... would just not implement this method if you don't care about exports (as that will cause only explicit exports to be exported).

Reply:
Will remove.